### ISSUE TITLE
Bdiv ordinal plot Umap 

### ISSUE THREAD

**Issue by shwetha141** [cmmr/rbiom#18]
Hello , I get this error whenever I run bdiv_ord_plot.

**Error in irlba::irlba(L, nv = n, nu = 0, maxit = iters) : 
  function 'as_cholmod_sparse' not provided by package 'Matrix'**

I thought it was my data but I get the same error for the  example data set hmp. I tried this site-https://github.com/satijalab/seurat/issues/8100

R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.4.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
 [1] parallel  grid      stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ape_5.7-1                       readxl_1.4.3                   
 [3] data.table_1.15.4               RColorBrewer_1.1-3             
 [5] heatmaply_1.5.0                 Hmisc_5.1-2                    
 [7] DT_0.33                         knitr_1.45                     
 [9] tsne_0.1-3.1                    GGally_2.2.1                   
[11] reshape2_1.4.4                  robustHD_0.8.0                 
[13] robustbase_0.99-2               perry_0.3.1                    
[15] corrplot_0.92                   gplots_3.1.3.1                 
[17] rstatix_0.7.2                   ComplexHeatmap_2.18.0          
[19] DESeq2_1.42.1                   ANCOMBC_2.4.0                  
[21] phyloseq_1.46.0                 glue_1.7.0                     
[23] viridis_0.6.5                   viridisLite_0.4.2              
[25] plotly_4.10.4                   ggpubr_0.6.0                   
[27] gridExtra_2.3                   hrbrthemes_0.8.7               
[29] rbiom_2.0.0.9120                BiocManager_1.30.22            
[31] vegan_2.6-4                     lattice_0.22-6                 
[33] permute_0.9-7                   lubridate_1.9.3                
[35] forcats_1.0.0                   stringr_1.5.1                  
[37] dplyr_1.1.4                     purrr_1.0.2                    
[39] readr_2.1.5                     tidyr_1.3.1                    
[41] tibble_3.2.1                    ggplot2_3.5.0                  
[43] tidyverse_2.0.0                 TreeSummarizedExperiment_2.10.0
[45] Biostrings_2.70.3               XVector_0.42.0                 
[47] SingleCellExperiment_1.24.0     SummarizedExperiment_1.32.0    
[49] Biobase_2.62.0                  GenomicRanges_1.54.1           
[51] GenomeInfoDb_1.38.8             IRanges_2.36.0                 
[53] MatrixGenerics_1.14.0           matrixStats_1.2.0              
[55] S4Vectors_0.40.2                BiocGenerics_0.48.1            

loaded via a namespace (and not attached):
  [1] R.methodsS3_1.8.2           gld_2.6.6                  
  [3] nnet_7.3-19                 TH.data_1.1-2              
  [5] vctrs_0.6.5                 energy_1.7-11              
  [7] digest_0.6.35               png_0.1-8                  
  [9] shape_1.4.6.1               proxy_0.4-27               
 [11] Exact_3.2                   registry_0.5-1             
 [13] ggrepel_0.9.5               httpcode_0.3.0             
 [15] MASS_7.3-60.0.1             fontLiberation_0.1.0       
 [17] httpuv_1.6.15               foreach_1.5.2              
 [19] withr_3.0.0                 xfun_0.43                  
 [21] survival_3.5-8              doRNG_1.8.6                
 [23] memoise_2.0.1               crul_1.4.0                 
 [25] ggbeeswarm_0.7.2            emmeans_1.10.1             
 [27] gmp_0.7-4                   systemfonts_1.0.6          
 [29] ragg_1.3.0                  tidytree_0.4.6             
 [31] zoo_1.8-12                  GlobalOptions_0.1.2        
 [33] gtools_3.9.5                R.oo_1.26.0                
 [35] DEoptimR_1.1-3              Formula_1.2-5              
 [37] promises_1.3.0              httr_1.4.7                 
 [39] rhdf5filters_1.14.1         rhdf5_2.46.1               
 [41] rstudioapi_0.16.0           generics_0.1.3             
 [43] base64enc_0.1-3             curl_5.2.1                 
 [45] zlibbioc_1.48.2             ScaledMatrix_1.10.0        
 [47] ca_0.71.1                   GenomeInfoDbData_1.2.11    
 [49] SparseArray_1.2.4           xtable_1.8-4               
 [51] ade4_1.7-22                 doParallel_1.0.17          
 [53] evaluate_0.23               S4Arrays_1.2.1             
 [55] hms_1.1.3                   irlba_2.3.5.1              
 [57] colorspace_2.1-0            magrittr_2.0.3             
 [59] later_1.3.2                 DECIPHER_2.30.0            
 [61] scuttle_1.12.0              class_7.3-22               
 [63] pillar_1.9.0                nlme_3.1-164               
 [65] iterators_1.0.14            decontam_1.22.0            
 [67] caTools_1.18.2              compiler_4.3.2             
 [69] beachmat_2.18.1             stringi_1.8.3              
 [71] biomformat_1.30.0           TSP_1.2-4                  
 [73] DescTools_0.99.54           dendextend_1.17.1          
 [75] minqa_1.2.6                 plyr_1.8.9                 
 [77] crayon_1.5.2                abind_1.4-5                
 [79] scater_1.30.1               ggtext_0.1.2               
 [81] locfit_1.5-9.9              bit_4.0.5                  
 [83] mia_1.10.0                  rootSolve_1.8.2.4          
 [85] sandwich_3.1-0              codetools_0.2-20           
 [87] multcomp_1.4-25             textshaping_0.3.7          
 [89] BiocSingular_1.18.0         slam_0.1-50                
 [91] e1071_1.7-14                lmom_3.0                   
 [93] GetoptLong_1.0.5            multtest_2.58.0            
 [95] mime_0.12                   MultiAssayExperiment_1.28.0
 [97] splines_4.3.2               circlize_0.4.16            
 [99] Rcpp_1.0.12                 sparseMatrixStats_1.14.0   
[101] cellranger_1.1.0            Rttf2pt1_1.3.12            
[103] gridtext_0.1.5              blob_1.2.4                 
[105] utf8_1.2.4                  clue_0.3-65                
[107] lme4_1.1-35.2               fs_1.6.3                   
[109] checkmate_2.3.1             DelayedMatrixStats_1.24.0  
[111] Rdpack_2.6                  expm_0.999-9               
[113] gsl_2.1-8                   ggsignif_0.6.4             
[115] estimability_1.5            Matrix_1.6-5               
[117] tzdb_0.4.0                  pkgconfig_2.0.3            
[119] tools_4.3.2                 cachem_1.0.8               
[121] rbibutils_2.2.16            RSQLite_2.3.6              
[123] DBI_1.2.2                   numDeriv_2016.8-1.1        
[125] fastmap_1.1.1               rmarkdown_2.26             
[127] scales_1.3.0                broom_1.0.5                
[129] patchwork_1.2.0             coda_0.19-4.1              
[131] ggstats_0.6.0               carData_3.0-5              
[133] rpart_4.1.23                farver_2.1.1               
[135] mgcv_1.9-1                  foreign_0.8-86             
[137] cli_3.6.2                   webshot_0.5.5              
[139] lifecycle_1.0.4             uwot_0.1.16                
[141] mvtnorm_1.2-4               bluster_1.12.0             
[143] backports_1.4.1             BiocParallel_1.36.0        
[145] timechange_0.3.0            gtable_0.3.4               
[147] rjson_0.2.21                CVXR_1.0-12                
[149] jsonlite_1.8.8              seriation_1.5.4            
[151] bitops_1.0-7                assertthat_0.2.1           
[153] bit64_4.0.5                 yulab.utils_0.1.4          
[155] BiocNeighbors_1.20.2        RcppParallel_5.1.7         
[157] R.utils_2.12.3              lazyeval_0.2.2             
[159] shiny_1.8.1.1               htmltools_0.5.8.1          
[161] gfonts_0.2.0                gdtools_0.3.7              
[163] RCurl_1.98-1.14             treeio_1.26.0              
[165] boot_1.3-30                 igraph_2.0.3               
[167] extrafontdb_1.0             R6_2.5.1                   
[169] labeling_0.4.3              Rmpfr_0.9-5                
[171] ggh4x_0.2.8                 cluster_2.1.6              
[173] rngtools_1.5.2              Rhdf5lib_1.24.2            
[175] nloptr_2.0.3                DirichletMultinomial_1.44.0
[177] DelayedArray_0.28.0         tidyselect_1.2.1           
[179] vipor_0.4.7                 htmlTable_2.4.2            
[181] xml2_1.3.6                  fontBitstreamVera_0.1.1    
[183] car_3.1-2                   rsvd_1.0.5                 
[185] munsell_0.5.1               KernSmooth_2.23-22         
[187] fontquiver_0.2.1            htmlwidgets_1.6.4          
[189] rlang_1.1.3                 extrafont_0.19             
[191] lmerTest_3.1-3              ggnewscale_0.4.10          
[193] fansi_1.0.6                 beeswarm_0.4.0

**Comment by dansmith01** [comment]
Hi @shwetha141 

Did any of the comments on https://github.com/satijalab/seurat/issues/8100 help to fix this for you?

**Comment by shwetha141** [comment]
Hi, I tried installing Matrix and irlba from source, it didn't work. Yet to work on the solution below.

This seems a recent change since I had no issues on 5th of April, when I wrote the code.  :(


_Thank you to both Cinzialo and zhuldyzhanzak! I have now solved the problem. Here is my solution: 1. Go to https://cran.r-project.org/bin/macosx/tools/, download an installer package [gfortran-12.2-universal.pkg](https://mac.r-project.org/tools/gfortran-12.2-universal.pkg)
2. In R: install.packages("Matrix", type = "source")
3. In R: install.packages("irlba", type = "source")_

**Comment by shwetha141** [comment]
Hi , this issue has been resolved, the following solution worked for me. Thank you.!

_Thank you to both Cinzialo and zhuldyzhanzak! I have now solved the problem. Here is my solution: 1. Go to https://cran.r-project.org/bin/macosx/tools/, download an installer package [gfortran-12.2-universal.pkg](https://mac.r-project.org/tools/gfortran-12.2-universal.pkg)
2. In R: install.packages("Matrix", type = "source")
3. In R: install.packages("irlba", type = "source")_
